util/reggen/clocking.py

```diff
@@ -49,6 +54,16 @@ def _asdict(self) -> Dict[str, object]:
         ret['primary'] = self.primary
         return ret
 
+    def get_base_name(self) -> str:
+        if not self.clock:
+            raise ValueError('Cannot extract clock base name from None clock')
+
+        match = re.match(r'^clk_([A-Za-z0-9_]+)_i', str(self.clock))
+        if match:
+            return match.group(1)
+        else:
+            return ""
```

Comment:
I think we expect all clocks to match `clk_.*_i`. Can we move that check to the `ClockingItem` constructor (raising an error on failure), and turn this into a simple getter?

Reply:
good idea. let me do that.